kernel: Reorder config files

hauke opened the pull request:
* mediatek: Remove double CONFIG_HARDEN_BRANCH_PREDICTOR
    
    CONFIG_HARDEN_BRANCH_PREDICTOR option is included two times in this
    configuration file. Remove one definition. On arm32 SoC it should not be
    needed.
    
    Fixes: 54f9744c820a ("treewide: disable spectre mitigation on unaffected Arm32 targets")

 * kernel: Reorder config files
    
    Reorder the kernel configuration files.
    
    I used this script to generate the commit, llama3.2 helped with the
    script:
    ```
    
    find_files=$(find target -type f -name 'config-*' -print)
    
    if [ -n "$find_files" ]; then
      echo "Found files:"
      for file in $find_files; do
        echo "Running kconfig.pl for $file"
        LANG=C ./scripts/kconfig.pl '+' "$file" /dev/null | sponge "$file"
      done
    else
      echo "No files named config-6.6 found."
    fi
    ```

hauke committed:
kernel: Reorder config files

Reorder the kernel configuration files.

This was done uisng:
./scripts/kconfig-reorder.sh

Link: https://github.com/openwrt/openwrt/pull/16743
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>

hauke committed:
mediatek: Remove double CONFIG_HARDEN_BRANCH_PREDICTOR

CONFIG_HARDEN_BRANCH_PREDICTOR option is included two times in this
configuration file. Remove one definition. On arm32 SoC it should not be
needed.

Fixes: 54f9744c820a ("treewide: disable spectre mitigation on unaffected Arm32 targets")
Link: https://github.com/openwrt/openwrt/pull/16743
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>

neheb commented on target/linux/generic/config-filter:
```diff
@@ -1,17 +1,17 @@
+# CONFIG_ARCH_(ENABLE|HAS|HAVE|INLINE|SUPPORTS|USE|WANT|STACKWALK)_.* is not set
```
is this right?

hauke replied:
I think it is no problem to move this.

Ansuel commented:
@hauke I also faced recently the task of cleaning a kernel config... Worth adding one of these utility script in maintainer-tools or scripts?

hauke committed:
scripts: Add script to reorder kernel config-* files

This script will reorder the content of all config-* files in the target
folder. It will also remove duplicates. It will not remove options
already defined in the generic configuration.

Link: https://github.com/openwrt/openwrt/pull/16743
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>

hauke commented:
I added the script in a separate commit.

Ansuel commented:
Thanks a lot very useful!
